### PR TITLE
docs: Update readme and samples to reflect separated packages

### PR DESCRIPTION
**Pull request by dazuma**
This removes most of the toplevel README content that discusses library usage. It just retains the getting-started examples, and points to the usage documentation for details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,68 +1,26 @@
...
+Each client provides:
 
-For Google Cloud Platform APIs such as Datastore, Cloud Storage or Pub/Sub, we recommend using [GoogleCloudPlatform/google-cloud-ruby](https://github.com/GoogleCloudPlatform/google-cloud-ruby) which is under active development.
+* A client object that connects to the HTTP/JSON REST endpoint for the service.
+* Ruby objects for data structures related to the service.
+* Integration with the googleauth gem for authentication using OAuth, API keys, and service accounts.
+* Control of retry, pagination, and timeouts.
 
-## Migrating from 0.8.x
+These client libraries are officially supported by Google, and are updated regularly to track changes to the service. However, they are considered legacy. A more modern client may be available for many Google services, especially Google Cloud Platform services such as Cloud Storage, Pub/Sub, and BigQuery. See the section below titled *Which client should I use?* for more information.
```
Comment (quartzmo) [inline README.md:12]: Use an anchor link to *Which client should I use?*?

File: README.md [hunk 2]
```diff
@@ -1,68 +1,26 @@
-# Google API Client [![Documentation](https://img.shields.io/badge/docs-Google%3A%3AApis-red.svg)](https://googleapis.dev/ruby/google-api-client/latest/Google/Apis.html) [![Gem Version](https://badge.fury.io/rb/google-api-client.svg)](https://badge.fury.io/rb/google-api-client)
+# Legacy REST Clients for Google APIs
```
Comment (quartzmo) [inline README.md:1]: Is it required to label all of these clients as "legacy"? I find it a bit confusing in those cases where no newer client exists.

**Review by quartzmo — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (dazuma) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (dazuma) [reply README.md]: I've been going back and forth on that. But you're right, it's probably premature to call them "legacy" at this point. I changed the label to "simple".

**Pull request merged**